src/components/PostDetails.tsx

```diff
@@ -66,29 +42,40 @@ export const PostDetails: React.FC<Props> = ({ post }) => {
         postId: post.id,
       });
 
-      setComments(currentComments => [...currentComments, newComment]);
-
-      // setComments([...comments, newComment]);
-      // works wrong if we wrap `addComment` with `useCallback`
-      // because it takes the `comments` cached during the first render
-      // not the actual ones
+      dispatch(addNewComment(newComment));
     } catch (error) {
-      // we show an error message in case of any error
-      setError(true);
+      dispatch(changeError());
```

Comment:
The 'changeError' action is dispatched without any specific error information. If possible, consider passing error details to provide more context for debugging or user feedback.